### DISCUSSION TITLE
Get last merged PR in main

### DISCUSSION THREAD

**Discussion by oana-marina** [community/community#134198] [Actions] [answered]
### Select Topic Area

Question

### Body

Is it possible to get the last merged PR on the main branch inside a workflow? I want to run test coverage during commit, and  fail the PR check if the test coverage is going down. For that I need the coverage from the last merged PR in main.

**Comment by AntA001** [comment, marked as answer]
Of course, what you are saying can very easily be done with github cli.

You setup the cli inside your workflow file, then you fetch the last merged PR you mentioned, then you run your tests. That will give you the current coverage with the merged PR, and after that you simply compare it with the previous coverage.

All the steps above can be done within the workflow itself, I can probably share a general outline if you are not sure how to do it.

And for the comparison even there you can write a simple js script to automate the comparison and include that in the workflow too, if you do not fancy doing it manually.

**Reply by oana-marina** [reply]
ok, I am really new to github actions so I am a bit stuck. I managed to run  
`gh pr list -s merged -B main -L 1 --repo ${{  github.repository }}`
inside the workflow and got a result, but then I can't figure out how to put the result in a variable and parse it so I can get just the ID. 
I tried lastMergedPR="$(gh pr list -s merged -B main -L 1 --repo ${{  github.repository }})" but I got an error...

**Reply by AntA001** [reply]
```
lastMergedPR=$(gh pr list -s merged -B main -L 1 --json number --jq '.[0].number')
echo "lastMergedPR=$lastMergedPR" >> $GITHUB_ENV
```

**Reply by AntA001** [reply]
If that works for you, I would appriciate if you can mark my ansswer as a reply too, I'm working towards the achievement haha.

If you need any more help let me know

**Reply by grandmas-favorite** [reply]
Nice one @AntA001!!! 😄
